### PR TITLE
fix #1717: all points must have at least one field

### LINKED ISSUE #1717: return error if write has no fields
We recently changed the data format of writes from `values` to `fields` in the code but not in the seed scripts.  This uncovered a bug.  Running an old seed script causes influxdb to add the measurement without any fields.  It should instead return an error if the write doesn't have any `fields`.

Recreate:

- start with clean database (`rm -rf ~/.influxdb`  deletes ALL data)
- run this script:  https://gist.github.com/dgnorton/d72f81c6c9f6a7a8df67

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -1369,6 +1369,13 @@ type Point struct {
 // WriteSeries writes series data to the database.
 // Returns the messaging index the data was written to.
 func (s *Server) WriteSeries(database, retentionPolicy string, points []Point) (uint64, error) {
+	// Make sure every point has at least one field.
+	for _, p := range points {
+		if len(p.Fields) == 0 {
+			return 0, fmt.Errorf("point %s has no fields", p.Name)
```
Comment (otoolep) [inline server.go]: Just return `ErrFieldsRequired`. In any event, `cpu` is a Measurement.

Comment (otoolep) [inline server.go]: To be clear, `ErrFieldsRequired` already exists in the source.

**Comment by pauldix** [comment]
Update the changelog and address @otoolep's comment. Other than that LGTM.

File: server.go (hunk 1 above)
Comment (otoolep) [inline server.go]: Also, why not do this at line 1403? We are already walking the points there. Let's not walk at the top, then walk again.

**Comment by otoolep** [comment]
Change makes sense, but it seems to me that the code should use the existing point-walking loop.

File: server.go (hunk 1 above)
Comment (dgnorton) [inline server.go]: At line 1403, the Measurements and ShardGroups have been created...they would have to be deleted.  Or, we'd have to move creation of measurements and shard groups inside the loop and created them one at a time.  Maybe that would give better performance?...not sure.

Comment (otoolep) [inline server.go]: OK, I see what you mean. Ironically `createMeasurementsIfNotExists` also loops through the points and I wrote that. 

Now that I see it, this function does a lot of stuff with a batch -- create shards, create series etc. It now seems to me you're on the right path, and that some basic validation of the batch is quite important before anything else happens. It kills me that we loop through the batch so many times. Oh well, we can profile it and see how it is, if it comes to it.

**Comment by otoolep** [comment]
+1 on this change now, it's pretty important that the batch is sane before anything else happens.

**Comment by otoolep** [comment]
+1, as long as the error code is fixed up, of course.

**Pull request merged**